src/sortedcache.c

```diff
@@ -207,6 +207,9 @@ int git_sortedcache_lockandload(git_sortedcache *sc, git_buf *buf)
 	int error, fd;
 	struct stat st;
 
+	if ((error = git_futils_filestamp_check_readonly(&sc->stamp, sc->path)) <= 0)
```

Comment:
This is very big optimization when resolving references in parallel from multiple threads. Without it there is no gain from multiple threads at all.